plugins/woocommerce/src/Admin/Features/MarketingRecommendations/Init.php

```diff
@@ -54,21 +56,60 @@ public static function get_specs() {
 		return $specs;
 	}
 
+	/**
+	 * Process specs.
+	 *
+	 * @param array|null $specs Marketing recommendations spec array.
+	 * @return array
+	 */
+	protected static function evaluate_specs( array $specs = null ) {
+		$suggestions = array();
+		$errors      = array();
+
+		foreach ( $specs as $spec ) {
+			try {
+				$suggestions[] = self::object_to_array( $spec );
```

Comment:
Even though this may not error since it's a simple operation, I'd like to be super defensive about processing any remote data, to ensure no fatals ever possible.